### PR TITLE
calculator: improved the code and added a simple user interface

### PR DESCRIPTION
**Pull request by christianbender**
I improved the code:  
- more efficient  in the ```evaluate(...)```  
- added a error management in the ```parser(...)``` and in the UI.  

And I added a simple user interface (command-line)

### CODE REVIEW

File: calculator/math_parser.py [hunk 1]
```diff
@@ -104,7 +108,8 @@ def evaluate(expression):
     """
     op_stack  = deque() # operator stack
     out_stack = deque() # output stack (values)
-    for token in parse(expression):
+    tokens = parse(expression) # calls the function only once!
+    for token in tokens:
```
Comment (goswami-rahul) [inline calculator/math_parser.py]: Actually, the expression to the right of `for` loop is evaluated only once. Hence no need for this.

File: calculator/math_parser.py [hunk 2]
```diff
@@ -89,8 +90,11 @@ def parse(expression):
             if len(current) > 0:
                 result.append(current)
                 current = ""
-            if i != ' ':
+            if (i in __operators__) or (i in __parenthesis__):
```
Comment (goswami-rahul) [inline calculator/math_parser.py]: no need for the `()` so you can remove them

File: calculator/math_parser.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
 
 from collections import deque
 import re
+import sys
```
Comment (goswami-rahul) [inline calculator/math_parser.py]: where is `sys` import used?

**Comment by goswami-rahul** [comment]
@christianbender  Good work. I have left some comments for you suggesting some changes.

File: calculator/math_parser.py [hunk 4]
```diff
@@ -122,3 +127,22 @@ def evaluate(expression):
...
+    """
+    print ("\t\tCalculator\n\n")
+    userInput = input("expression or exit: ")
+    while userInput != "exit":
+        try:
+            print ("The result is {0}".format(evaluate(userInput)))
+        except Exception:
+            print ("invalid syntax!")
+        userInput = input("expression or exit: ")
+    print ("program end")
```
Comment (goswami-rahul) [inline calculator/math_parser.py]: please remove the space between `print` and `(` at all usages.

File: calculator/math_parser.py (hunk 1 above)
Comment (christianbender) [reply calculator/math_parser.py]: Ok, I understand.

File: calculator/math_parser.py (hunk 3 above)
Comment (christianbender) [reply calculator/math_parser.py]: Ohh, upps I used this for exit the program. I changed my mind. I remove it.

File: calculator/math_parser.py (hunk 4 above)
Comment (christianbender) [reply calculator/math_parser.py]: OK,

**Comment by christianbender** [comment]
@goswami-rahul  I have changed my code across your request.

**Pull request merged**